src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala

```diff
@@ -174,21 +174,32 @@ class LibraryApiServiceSpec extends BaseServiceSpec with LibraryApiService with
       }
     }
     "when updating fields for a published workspace" - {
-      "should republish the workspace" in {
+      "should republish the workspace for writer" in {
         val content = HttpEntity(ContentTypes.`application/json`, testLibraryMetadata)
         new RequestBuilder(HttpMethods.PUT)(setMetadataPath("publishedwriter"), content) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
           status should equal(OK)
           assert(this.searchDao.indexDocumentInvoked, "indexDocument should have been invoked")
         }
       }
       "should be forbidden when reader" in {
-        new RequestBuilder(HttpMethods.POST)(publishedPath("publishedreader")) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+        val content = HttpEntity(ContentTypes.`application/json`, testLibraryMetadata)
+        new RequestBuilder(HttpMethods.PUT)(setMetadataPath("publishedreader"), content) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
           status should equal(Forbidden)
+          assert(!this.searchDao.indexDocumentInvoked, "indexDocument should not have been invoked")
         }
       }
       "should be allowed when writer" in {
-        new RequestBuilder(HttpMethods.POST)(publishedPath("publishedwriter")) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+        val content = HttpEntity(ContentTypes.`application/json`, testLibraryMetadata)
+        new RequestBuilder(HttpMethods.PUT)(setMetadataPath("publishedwriter"), content) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
           status should equal(OK)
+          assert(this.searchDao.indexDocumentInvoked, "indexDocument should have been invoked")
+        }
+      }
+      "should be allowed when reader and catalog" in {
+        val content = HttpEntity(ContentTypes.`application/json`, testLibraryMetadata)
+        new RequestBuilder(HttpMethods.PUT)(setMetadataPath("publishedreadercatalog"), content) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+          status should equal(OK)
+          assert(this.searchDao.indexDocumentInvoked, "indexDocument should have been invoked")
```

Comment:
as with my previous comment, this is wrong. The `userAttributeOne`, `userAttributeTwo`, and `description` in the `testLibraryMetadata` should NOT be updated if you only have reader.